ql/src/java/org/apache/hadoop/hive/ql/plan/PlanUtils.java

```diff
@@ -412,36 +411,63 @@ public static TableDesc getTableDesc(CreateViewDesc crtViewDesc, String cols, St
     TableDesc ret;
 
     try {
-      Class serdeClass = JavaUtils.loadClass(crtViewDesc.getSerde());
-      ret = getTableDesc(serdeClass, new String(LazySerDeParameters.DefaultSeparators), cols,
-          colTypes, false,  false);
+      HiveStorageHandler storageHandler = null;
+      if (crtViewDesc.getStorageHandler() != null) {
+        storageHandler = HiveUtils.getStorageHandler(
+                SessionState.getSessionConf(), crtViewDesc.getStorageHandler());
+      }
+
+      Class<? extends Deserializer> serdeClass = LazySimpleSerDe.class;
```

Comment:
Doesn't seem like a good idea to use lazysimple if none is specified. Shall we throw in some cases.

Reply:
This code is also similar to the one for the normal table descriptors, so we use the same strategy. I will consolidate the code here too.